src/Microsoft.ML.Data/Evaluators/Metrics/ConfusionMatrix.cs

```diff
@@ -0,0 +1,122 @@
+    /// </summary>
+    public sealed class ConfusionMatrix
+    {
+        /// <summary>
+        /// The calculated value of <a href="https://en.wikipedia.org/wiki/Precision_and_recall#Precision">precision</a> for each class.
+        /// </summary>
+        public ImmutableArray<double> PerClassPrecision { get; }
+
+        /// <summary>
+        /// The calculated value of <a href="https://en.wikipedia.org/wiki/Precision_and_recall#Recall">recall</a> for each class.
+        /// </summary>
+        public ImmutableArray<double> PerClassRecall { get; }
+
+        /// <summary>
+        /// The confsion matrix counts for the combinations actual class/predicted class.
+        /// The actual classes are in the rows of the table, and the predicted classes in the columns.
+        /// </summary>
+        public ImmutableArray<ImmutableArray<double>> Counts { get; }
+
+        /// <summary>
+        /// The indicators of the predicted classes.
+        /// It might be the classes names, or just indices of the predicted classes, if the name mapping is missing.
+        /// </summary>
+        public int NumberOfPredictedClasses { get; }
+
+        /// <summary>
+        /// The <see cref="DataViewSchema.Annotations"/> associated with the Confusion Matrix counts.
+        /// It contains information about the predicted classes, if that information is available.
+        /// It can be the classes names, or their indices.
+        /// </summary>
+        public DataViewSchema.Annotations ClassIndicators { get; }
+
+        /// <summary>
+        /// The indicators of the predicted classes.
+        /// It might be the classes names, or just indices of the predicted classes, if the name mapping is missing.
+        /// </summary>
+        internal IReadOnlyList<ReadOnlyMemory<char>> PredictedClassesIndicators;
+
+        internal readonly bool IsSampled;
+        internal readonly bool IsBinary;
+
+        private readonly IHost _host;
+
+        /// <summary>
+        /// The confusion matrix as a structured type, built from the counts of the confusion table idv.
+        /// </summary>
+        /// <param name="host">The IHost instance. </param>
+        /// <param name="precision">The values of precision per class.</param>
+        /// <param name="recall">The vales of recall per class.</param>
+        /// <param name="confusionTableCounts">The counts of the confusion table. The actual classes values are in the rows of the 2D array,
+        /// and the counts of the predicted classes are in the columns.</param>
+        /// <param name="labelNames">The predicted classes names, or the indexes of the classes, if the names are missing.</param>
+        /// <param name="isSampled">Whether the classes are sampled.</param>
+        /// <param name="isBinary">Whether the confusion table is the result of a binary classification. </param>
+        /// <param name="classIndicators">The Annotations of the Count column, in the confusionTable idv.</param>
```

Comment:
>confusionTable idv [](start = 86, length = 18)

`<paramref name="confusionTableCounts"/>`?

Reply:
I meant the Annotations in the original IDV that gets created by the evaluator. Expanding the comment with the above note. 

---
In reply to: [275468277](https://github.com/dotnet/machinelearning/pull/3250#discussion_r275468277) [](ancestors = 275468277)

Reply:
Actually removed the param, so description went away too. 

---
In reply to: [275527727](https://github.com/dotnet/machinelearning/pull/3250#discussion_r275527727) [](ancestors = 275527727,275468277)